4 slim query

ndaelman-hu opened the pull request:
closes #4

Commit:
Add non-merged correction

Commit:
Fix indentation + typo

ndaelman-hu commented:
> In general, you can use the [NOMAD API](https://nomad-lab.eu/prod/rae/docs/api.html) to grab particular archive entries from the repository or to search the repository for entries with certain attributes. This can be achieved in python using the `requests` module.

While it is good to point to the documentation, the most useful resource imho is the [API schema](https://nomad-lab.eu/prod/v1/staging/api/v1/extensions/docs#/), since it lists ALL options.
I would at least provide a link to it, saying that one is the beginners introduction, the other is the full resource list.
I would also integrate it into the tutorial so users feel surer relying on it.

Btw, depending on the technical level of the audience, I would drop the term RESTful API somewhere (it's also in your links). For those familiar, it will give them the reference frame to know how to navigate.

**Note:** I can apply this too in this PR, if you want.

JFRudzinski commented on docs/Tutorial-3_Extracting_Data_and_Trajectory_Analysis.md:
```diff
@@ -15,13 +15,11 @@ conda install "ipywidgets <8" -c conda-forge
 !!! note "NOTE"
-
-        You may have to restart vscode or other editor after updating `ipywidgets`.
+    You may have to restart your IDE after updating `ipywidgets`.
```
I don't think this is a term that everyone necessarily knows. I would rather use a less technical term or add a clarification like "i.e., your code editor" or something like that

ndaelman-hu replied:
Okay, I'll put "code editor (IDE)"

JFRudzinski replied:
great :+1:

JFRudzinski reviewed (approved):
Great, thanks. Just one comment about the IDE, see below...

ndaelman-hu commented:
```python
  conda create -n "CECAM_tutorial" python==3.9
  conda activate CECAM_tutorial
  pip install nomad-lab
  conda install nglview
  conda install "ipywidgets <8" -c conda-forge
  ```
How about we provide a setup file that they download (like requirements.txt). They can always take a look there if they want to know the necessary packages.

JFRudzinski commented:
> ```python
>  conda create -n "CECAM_tutorial" python==3.9
>  conda activate CECAM_tutorial
>  pip install nomad-lab
>  conda install nglview
>  conda install "ipywidgets <8" -c conda-forge
> ```
> 
> How about we provide a setup file that they download (like requirements.txt). They can always take a look there if they want to know the necessary packages.

yes, I was thinking of doing this, I just didn't get that far. I will add this.

ndaelman-hu commented:
> > ```python
> >  conda create -n "CECAM_tutorial" python==3.9
> >  conda activate CECAM_tutorial
> >  pip install nomad-lab
> >  conda install nglview
> >  conda install "ipywidgets <8" -c conda-forge
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > How about we provide a setup file that they download (like requirements.txt). They can always take a look there if they want to know the necessary packages.
> 
> yes, I was thinking of doing this, I just didn't get that far. I will add this.

I would do it else, but then I have to install conda again...
Anyhow, if you have a working environment (but just the bare minimum to make the notebook work), you can dump the info via
`conda list -e > requirements.txt` or `conda env export > requirements.txt`
I will also add it to our requirements (virtualenv) for the mkdocs.

JFRudzinski commented:
> > In general, you can use the [NOMAD API](https://nomad-lab.eu/prod/rae/docs/api.html) to grab particular archive entries from the repository or to search the repository for entries with certain attributes. This can be achieved in python using the `requests` module.
> 
> While it is good to point to the documentation, the most useful resource imho is the [API schema](https://nomad-lab.eu/prod/v1/staging/api/v1/extensions/docs#/), since it lists ALL options. I would at least provide a link to it, saying that one is the beginners introduction, the other is the full resource list. I would also integrate it into the tutorial so users feel surer relying on it.
> 
> Btw, depending on the technical level of the audience, I would drop the term RESTful API somewhere (it's also in your links). For those familiar, it will give them the reference frame to know how to navigate.
> 
> **Note:** I can apply this too in this PR, if you want.

Yes, this sounds like a good idea to me. Good you go ahead and add/change it and I will just check it when I review the entire thing tomorrow.

Commit:
Add 2nd dependency setup for jupyter lab

Commit:
Update notebook requirements

Commit:
In the Notebook:
- Rewrite introdcution to the NOMAD API
- Add timer and other functionalities
- Add basic code for filtered download
+ TODO: expand on `post` and processing filtered data

In draft Notebook: try out some queries and archive conversion. Can be deleted later on

ndaelman-hu committed:
Show filtered query and `post`

ndaelman-hu committed:
Polish previous writing + remove ambiguous section

ndaelman-hu committed:
Fill in missing file size

JFRudzinski commented:
@ndaelman-hu I did not realize until now these changes you made to the notebook. These are great, but I need to do some work then to integrate them. I would like to start this ASAP. I suggest we merge #7 and this PR and then I can start working on this integration and refining things...everything is a bit fractured atm for me to work on. Is this possible, or do you still have a lot to do on this PR?

ndaelman-hu committed:
Relax some of the dependency versions

ndaelman-hu committed:
Add code for last exercise: extract data from multiple archives

ndaelman-hu committed:
Add story and explanation for extra exercise

ndaelman-hu committed:
Divide tutorial more clearly into chapters / exercises + polish exercise 6

ndaelman-hu commented:
So, while I removed your part on extracting the positions, velocities and box limits, I reintroduced **velocity extraction in the extra (last) exercise**. There, I make the reader query and filter **multiple archives** at once, comparing their velocity distribution. Also nice, I use the settings from scenario 1 in part I, so in a sense they come **full circle.**

I will see to merge this tonight, but @JFRudzinski if you want the comparison extended, I'll open a new mini-PR.

ndaelman-hu commented:
Also, your first example of the RDF plot is not water-water, but hexane-hexane... I've corrected this now.

ndaelman-hu commented:
> Also, your first example of the RDF plot is not water-water, but hexane-hexane... I've corrected this now.

The same goes for the protein mentioned afterwards...

ndaelman-hu committed:
Final check and polish + rectify water-water rdf with hex-hex

ndaelman-hu committed:
Remove testing notebook

ndaelman-hu commented:
> > Also, your first example of the RDF plot is not water-water, but hexane-hexane... I've corrected this now.
> 
> The same goes for the protein mentioned afterwards...

Seems that you already fixed this yourself in part 4.

ndaelman-hu committed:
Merge remote-tracking branch 'origin/main' into 4-slim-query
